### PR TITLE
Rename `fn_args_layout` -> `fn_params_layout`

### LINKED ISSUE #4149: Rename `fn_args_layout` config option
We previously (#3803) refactored the `arg` name to `param` internally within rustfmt  where appropriate to reflect the upstream changes (rust-lang/rust#62426), but did not change the public/external-facing [`fn_args_layout` config option](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#fn_args_layout) because that would have been a breaking change.

However, this config option is indeed used as part of the formatting of the parameters in a fn signature, not in the formatting of args in a fn call. As such this should also be renamed as part of the major version release (`fn_params_layout` seems  like a reasonable name)

@topecongiro I've added this to the 2.0 milestone per our previous discussion in https://github.com/rust-lang/rustfmt/pull/3803#issuecomment-533901176, but of course feel free to change that if you think we should defer it.

Below are the steps involved for anyone interested in working on this:
* Rename the configuration option and update the help text in `rustfmt-core/rustfmt-lib/src/config.rs`
https://github.com/rust-lang/rustfmt/blob/7194d529467cf7748bd0031ecf35ae8917f98992/rustfmt-core/rustfmt-lib/src/config.rs#L111-L112
    * make sure to update the corresponding test in that file too https://github.com/rust-lang/rustfmt/blob/master/rustfmt-core/rustfmt-lib/src/config.rs#L556
* Rename the corresponding test files/update config references under `rustfmt-core/rustfmt-lib/tests/{source,target}`
   * For example, rename [tests/source/fn_args_layout-vertical.rs](https://github.com/rust-lang/rustfmt/blob/master/rustfmt-core/rustfmt-lib/tests/source/fn_args_layout-vertical.rs)  (and the same for the matching file under `tests/target`)to `fn_params_layout-vertical.rs` and the inline config to be `// rustfmt-fn_params_layout: Vertical`
    * Same thing for [`tests/{source,target}/configs/fn_args_layout`]( https://github.com/rust-lang/rustfmt/tree/master/rustfmt-core/rustfmt-lib/tests/source/configs/fn_args_layout), rename the directory and update the inline config in the files with the new config option name
https://github.com/rust-lang/rustfmt/blob/7194d529467cf7748bd0031ecf35ae8917f98992/rustfmt-core/rustfmt-lib/tests/source/fn_args_layout-vertical.rs#L1-L2

* Rename/update the [corresponding section in Configurations.md](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#fn_args_layout)

### PR DESCRIPTION
**Pull request by ytmimi**
`fn_args_layout` was renamed `fn_params_layout` to better communicate that it affects the layout of parameters in function signatures and not the layout of arguments in function calls.

Because the `fn_args_layout` is a stable option the renamed option is also stable, however users who set `fn_args_layout` will get a warning message letting them know that the option has been renamed.

Closes #4149

### CODE REVIEW

**Comment by ytmimi** [comment]
This would backport #4163

File: src/config/config_type.rs [hunk 1]
```diff
@@ -400,6 +409,17 @@ macro_rules! create_config {
                 }
             }
 
+            fn set_fn_args_layout(&mut self) {
+                if self.was_set().fn_args_layout() {
+                    eprintln!(
+                        "Warning: the `fn_args_layout` option was renamed to `fn_params_layout`."
```
Comment (calebcartwright) [inline src/config/config_type.rs:415]: Can we also mention that the old option has been deprecated, similar to what we do for other such renames?

Comment (ytmimi) [reply src/config/config_type.rs:415]: Absolutely! I'll make sure to include those details.

File: Configurations.md [hunk 2]
```diff
@@ -645,7 +645,7 @@ trailing whitespaces.
 
 ## `fn_args_layout`
 
-Control the layout of arguments in a function
+This option has been renamed to `fn_params_layout` to better communicate that it affects the layout of parameters in function signatures.
```
[the remark this answers is not in the document]
Comment (ytmimi) [reply Configurations.md:648]: I'll update the docs to include the deprecation notice!

**Comment by calebcartwright** [comment]
Do you think it would be helpful to include some textual description for each variant in the corresponding section on the config docs? The code snippets are fine, but I think a sentence or two that captures behavior which may be non-obvious (e.g. there must be 2+ params for...) would be helpful.

**Comment by ytmimi** [comment]
Good call! I'm all for updating the descriptions to include as much detail as possible. I'll make those tweak sometime later this week

**Comment by calebcartwright** [comment]
> Good call! I'm all for updating the descriptions to include as much detail as possible. I'll make those tweak sometime later this week

Excellent, thank you. I realize that was an add-on request to the original scope of this PR so I'm going to go ahead and merge. Feel free to submit a follow up with proposed doc enhancements

**Pull request merged**